packages/mongoose/src/generateModel.ts

```diff
@@ -32,15 +33,16 @@ const DEFAULT_SCHEMA_OPTIONS = { timestamps: true, toJSON: { virtuals: true }, t
  */
 export const generateSchema = <T>(
 	schemaType: ClassType,
-	options: SchemaOptions = DEFAULT_SCHEMA_OPTIONS
+	options?: SchemaOptions
 ): Schema<T> => {
 	const recursivelyGenerateSchema = <U>(
 		// eslint-disable-next-line no-shadow
 		schemaType: ClassType,
 		// eslint-disable-next-line no-shadow
-		options: SchemaOptions | null = DEFAULT_SCHEMA_OPTIONS,
+		options?: SchemaOptions,
 		forceCreateSchema?: boolean
 	): RecursivelyGenerateSchemaReturnType<U> => {
+		const mergedOptions = options ? deepmerge(DEFAULT_SCHEMA_OPTIONS, options) : null;
```

Comment:
not sure I agree with this logic.
We are saying to merge the 'options' argument with the DEFAULT_SCHEMA_OPTIONS, only if the 'options' argument is set.

Reply:
I agree. I think now it should be better